### PR TITLE
docs: optimize code block structure

### PR DESCRIPTION
**Pull request by chenxsan**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

A follow-up pull request as we discussed in https://github.com/eslint/eslint/pull/16636#issuecomment-1353732929. Thanks @amareshsm for raising that issue which was introduced by my pull request.

#### What changes did you make? (Give an overview)

1. refractor syntax highlighting code to minimize changes of CSS.
2. Add `font-variant-ligatures: none;` rule to fix a UI issue under firefox reported https://github.com/eslint/eslint/pull/16671 by @Tanujkanti4441 
3. closes https://github.com/eslint/eslint/issues/16676

Here's a diff image (a very long one) before and after this pull request tested under firefox, so hopefully it won't break anything else other than fixing the language switcher style and number alignment issue:

And here's the page reported with [broken line number before](https://github.com/eslint/eslint/issues/16042): https://deploy-preview-16669--docs-eslint.netlify.app/rules/function-paren-newline#rule-details, no regression here.

### CODE REVIEW

File: docs/src/_plugins/md-syntax-highlighter.js
Comment (chenxsan) [inline docs/src/_plugins/md-syntax-highlighter.js:66]: We don't need wrapper now.

File: docs/src/_plugins/md-syntax-highlighter.js
Comment (chenxsan) [inline docs/src/_plugins/md-syntax-highlighter.js:80]: `markdown-it` will append a `\n` for highlighted result https://github.com/markdown-it/markdown-it/blob/1529ff4944329eb8080a8558dab85164b6c212cd/lib/renderer.js#L59

File: docs/src/assets/scss/syntax-highlighter.scss
Comment (chenxsan) [inline docs/src/assets/scss/syntax-highlighter.scss:48]: Calculated according to the original implement.

**Comment by harish-sethuraman** [comment]
Trying to re run tests 😅 random failures

File: docs/src/_plugins/md-syntax-highlighter.js
Comment (amareshsm) [inline docs/src/_plugins/md-syntax-highlighter.js:50]: `highlighter` fn returns a `string`. Can u mention that in js docs?

**Review by harish-sethuraman — APPROVED** [review]
I verified for `js | javascript`, `typescript`, `shell`, `json` `xml` fences in no order and it seemed to work fine ✅.
Lets just make sure we don't break anything else as well😄

**Review by kecrily — APPROVED** [review]
LGTM

**Review by amareshsm — APPROVED** [review]
LGTM

**Comment by amareshsm** [comment]
Pls, rebase with the main branch.

**Review by nzakas — APPROVED** [review]
LGTM. Thank you!

Everyone - if you approve a PR and don't merge it, please explain why. I see three approvals but no one merged it and I don't know if there's a reason.

**Comment by amareshsm** [comment]
> LGTM. Thank you!
> 
> Everyone - if you approve a PR and don't merge it, please explain why. I see three approvals but no one merged it and I don't know if there's a reason.

Just waiting for the code owner review from `eslint/eslint-team`.

**Pull request merged**
